metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -302,3 +385,185 @@ func getHostSystem(ctx context.Context, c *vim25.Client, ref types.ManagedObject
+	for _, vm := range vms {
+		vmMap[vm.Reference().Value] = vm
+	}
+	return vmMap
+}
+
+type resourcePoolInfo struct {
+	InventoryPath string
+	Name          string
+	ref           types.ManagedObjectReference
+}
+
+func (m *MetricSet) getResourcePoolMap(ctx context.Context, c *vim25.Client, mgr *view.Manager) (map[string]resourcePoolInfo, error) {
+	// Create a single ContainerView for ResourcePool types
+	r, err := mgr.CreateContainerView(ctx, c.ServiceContent.RootFolder, []string{"ResourcePool"}, true)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Destroy(ctx) // Simplified error handling, as the error is not critical
+
+	// Retrieve all resource pools in the cluster for later mapping
+	var rps []mo.ResourcePool
+	err = r.Retrieve(ctx, []string{"ResourcePool"}, []string{"summary"}, &rps)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a map to easily look up resource pools by reference
+	rpMap := make(map[string]resourcePoolInfo, len(rps))
+	for _, rp := range rps {
+		summary := rp.Summary.GetResourcePoolSummary()
+		rpInfo := resourcePoolInfo{
+			InventoryPath: "", // Placeholder, as we don't have the InventoryPath directly
+			Name:          summary.Name,
+		}
+		rpMap[rp.Reference().Value] = rpInfo
+	}
+
+	// Now, we need to fill in the InventoryPath for each resource pool
+	finder := find.NewFinder(c, false)
+	dcList, err := finder.DatacenterList(ctx, "*")
+	if err != nil {
+		return nil, err
+	}
+	for _, dc := range dcList {
+		if m.DataCenters != nil && !contains(m.DataCenters, dc.Name()) {
+			continue
+		}
+		finder.SetDatacenter(dc)
+		pools, err := finder.ResourcePoolList(ctx, "*")
+		if err != nil {
+			return nil, fmt.Errorf("unable to find resource pools for %s due to %w", dc.Name(), err)
+		}
+		for _, pool := range pools {
+			if m.ResourcePools != nil && !contains(m.ResourcePools, pool.InventoryPath) {
+				m.Logger().Debugf("skipping resource pool %s as it is not in the list of included resource pools", pool.InventoryPath)
+				continue
+			}
+			if rpInfo, exists := rpMap[pool.Reference().Value]; exists {
+				rpInfo.InventoryPath = pool.InventoryPath
+				rpInfo.ref = pool.Reference()
+				rpMap[pool.Reference().Value] = rpInfo // Update the map with the InventoryPath
+			}
+		}
+	}
+
+	m.Logger().Info("retrieved resource pool list")
+	return rpMap, nil
+}
+
+// this is currently unused since it's a very expensive call, but if we could cache it it would be nice to have
+func getVMInventoryPath(ctx context.Context, c *vim25.Client, vm mo.VirtualMachine) (string, error) {
+	// Create a property collector
+	pc := property.DefaultCollector(c)
+
+	// Traverse up the inventory tree to construct the full path
+	var pathElements []string
+	ref := vm.Reference()
+	for ref.Type != "Datacenter" {
+		var entity mo.ManagedEntity
+		err := pc.RetrieveOne(ctx, ref, []string{"name", "parent"}, &entity)
+		if err != nil {
+			return "", err
+		}
+		pathElements = append([]string{entity.Name}, pathElements...)
```

Comment:
any specific reason to not to do: `pathElements = append(pathElements, []string{entity.Name})`?

Reply:
each iteration of the loop needs to add the new path element to the beginning of the slice so when they are joined into a string they make a correctly formatted folder path to the VM.  it would also work to instead append the name at the end of the slice and then reverse it before doing the join if that would be cleaner/fit the usual style guidelines better.